client/tokenized-payment-request/frontend-utils.js

```diff
@@ -107,3 +126,112 @@ export const displayLoginConfirmationDialog = ( paymentRequestType ) => {
 		)?.redirect_url;
 	}
 };
+
+export const getErrorMessageFromNotice = ( notice ) => {
+	const div = document.createElement( 'div' );
+	div.innerHTML = notice.trim();
+	return div.firstChild ? div.firstChild.textContent : '';
+};
+
+/**
+ * Transforms the cartData object to the format expected by the Store API. cartData is coming to the blocks Payment Request method
+ * in two different formats: from the canMakePayment function and from the content component. This function takes in either format
+ * and transforms it into the format expected by the Store API.
+ *
+ * @param {Object|null} cartDataCanMakePayment cartData from canMakePayment function.
+ * @param {Object|null} cartDataContent cartData from content component.
+ * @return {Object} Cart totals object.
+ */
+export const transformCartDataForStoreAPI = (
+	cartDataCanMakePayment,
+	cartDataContent
+) => {
+	let mappedCartData = {};
+
+	if ( cartDataCanMakePayment ) {
+		mappedCartData = {
+			...cartDataCanMakePayment,
+			items: cartDataCanMakePayment.cart.cartItems,
+			totals: cartDataCanMakePayment.cartTotals,
+			needs_shipping: cartDataCanMakePayment.cartNeedsShipping,
+			shipping_rates: cartDataCanMakePayment.cart.shippingRates,
+		};
+	}
+
+	if ( cartDataContent ) {
+		mappedCartData = {
+			items: cartDataContent.cartItems,
+			totals: constructCartDataContentTotals( cartDataContent ),
+			needs_shipping: cartDataContent.needsShipping,
+			shipping_rates: cartDataContent.shippingRates,
+			extensions: cartDataContent.extensions,
+		};
+	}
+
+	return mappedCartData;
+};
+
+function constructCartDataContentTotals( cartDataContent ) {
```

Comment:
```suggestion
/**
 * Transforms totals from cartDataContent into format expected by the Store API.
 *
 * @param {Object} cartDataContent cartData from content component
 * @return {Object} Cart totals object for Store API
 */
const constructCartDataContentTotals = ( cartDataContent ) => {
```
Pedantic comment, but can we define these functions more consistently and add a doc comment pls to align with the other functions in this file? (May need to lift this up higher in this file, before this function is used.) 🙏

Reply:
Same as below - [bcd0021](https://github.com/Automattic/woocommerce-payments/pull/9150/commits/bcd00217a15119ebb7d8718de1a03cc33430df0e)